custom/icds/data_management/static/data_management/js/data_management.js

```diff
@@ -1,9 +1,94 @@
 hqDefine('data_management/js/data_management', [
     'jquery',
+    'knockout',
+    'underscore',
+    "hqwebapp/js/initial_page_data",
+    'hqwebapp/js/components.ko',    // pagination widget
+    'hqwebapp/js/knockout_bindings.ko', // for modals
     'jquery-ui/ui/datepicker',
-], function ($) {
+], function ($, ko, _, initialPageData) {
     'use strict';
+
+    var dataManagementRequest = function (name, dbAlias, createdAt, initiatedBy, startedOn, endedOn, startDate,
+        endDate, status, statusText, error) {
+        var self = {};
+        self.name = name;
+        self.dbAlias = dbAlias;
+        self.initiatedBy = initiatedBy + gettext(' On ') + createdAt;
+        self.startedOn = startedOn;
+        self.endedOn = endedOn;
+        if (startDate) {
+            self.dateRange = startDate + '-' + endDate;
+        } else {
+            self.dateRange = '';
+        }
+        self.statusText = statusText;
+        if (status === 3) {
```

Comment:
Can `3` be a constant instead of a magic number?

Reply:
It is a constant under model `DataManagementRequest` so you mean it to JS using InitialPageData?

Reply:
Or create the same named constant on the js side, just so this code is easier to read.